frontend/src/App.tsx

```diff
@@ -65,6 +68,109 @@ const App = () => {
   const [needsRefresh, setNeedsRefresh] = useState(false)
   const checkRefreshDuration = 600000 // (10 min)
 
+  const [currRoster, setCurrRoster] = useState<string>('')
+  const [hasLoadedCurrRoster, setHasLoadedCurrRoster] = useState<boolean>(false)
+
+  const loadCurrRoster = () => {
+    axios.get(`${API_ROOT}${SETTINGS_API}/semester/current`).then(
+      (req) => {
+        setCurrRoster(req.data)
+        setHasLoadedCurrRoster(true)
+      },
+      (error) => {
+        console.log(error)
+        setNetworkError(error.message)
+      }
+    )
+  }
+
+  const changeCurrRoster = async (event: SelectChangeEvent) => {
+    // function to only open the survey of the semester
+    setCurrRoster(event.target.value)
+    await axios.post(`${API_ROOT}${SETTINGS_API}/semester/current`, {
+      semester: event.target.value,
+    })
+  }
+
+  const [surveyState, setSurveyState] = useState<boolean>(false)
+  const [hasLoadedSurveyState, setHasLoadedSurveyState] = useState<boolean>(
+    false
+  )
+
+  const loadSurveyState = () => {
+    axios.get(`${API_ROOT}${SETTINGS_API}/semester/survey`).then(
+      (req) => {
+        setSurveyState(req.data)
+        setHasLoadedSurveyState(true)
+      },
+      (error) => {
+        console.log(error)
+        setNetworkError(error.message)
+      }
+    )
+  }
+
+  const changeSurveyAvailability = async () => {
+    axios.post(`${API_ROOT}${SETTINGS_API}/semester/survey`, {
+      surveyOpen: !surveyState,
+    })
+    setSurveyState(!surveyState)
+  }
+
+  const [administrators, setAdministrators] = useState<Admin[]>([])
+  const [
+    hasLoadedAdministrators,
+    setHasLoadedAdministrators,
+  ] = useState<boolean>(false)
+
+  const loadAdministrators = () => {
+    axios.get(`${API_ROOT}/admin`).then(
+      (req) => {
+        setAdministrators(req.data)
+        setHasLoadedAdministrators(true)
+      },
+      (error) => {
+        console.log(error)
+        setNetworkError(error.message)
+      }
+    )
+  }
+
+  const removeAdmin = (admin: Admin) => {
+    axios
+      .delete(`${API_ROOT}/admin`, { data: admin })
+      .then(() => {
+        administrators.forEach((email, index) => {
+          if ((email.email = admin.email)) {
+            administrators.splice(index, 1)
+            loadAdministrators()
+          }
+        })
+      })
+      .catch((err) => console.log(err))
+  }
+
+  // TODO: allow to edit admin information
+  const editAdmin = () => {}
```

Comment:
This should be changed to `add admin` and will be ready once Justin merges his PR.